tests/update/updates_packagekit_kde.pm

```diff
@@ -27,22 +27,21 @@ sub run {
     select_console 'x11', await_console => 0;
     ensure_unlocked_desktop;
     turn_off_kde_screensaver;
-
     my @updates_installed_tags = qw(updates_none updates_available updates_available-tray);
     assert_screen [qw(updates_available-tray tray-without-updates-available)];
     if (match_has_tag 'updates_available-tray') {
-        assert_and_click("updates_available-tray");
-
         # First update package manager, then packages, then bsc#992773 (2x)
         while (1) {
-            assert_and_click_until_screen_change('updates_click-install');
-
-            # Wait until installation starts, intended to time out
-            wait_still_screen(stilltime => 4, timeout => 5);
-
-            # Wait until installation is done
-            assert_screen \@updates_installed_tags, 3600;
+            wait_screen_change {
+                assert_and_click_until_screen_change("updates_available-tray");
+            };
+            wait_screen_change {
+                assert_and_click_until_screen_change('updates_click-install');
+            };
 
+            # Wait until installation is done.
+            do { } while (check_screen([qw(updates_installed_popup pkit_installing_state updates_waiting)]));
```

Comment:
That looks bad. This will execute `check_screen` as often as possible in a loop until it matches but also there is no timeout meaning if there is no match at all then this will eventually run into a openQA job timeout and make the job timeout_exceeded, not being able to execute post_fail_hook and alike . A better variant would be

```suggestion
            assert_screen [qw(updates_installed_popup pkit_installing_state updates_waiting)], timeout => 3600, no_wait => 1;
```

`no_wait => 1` meaning "run the screen check as fast as possible". Only do that to check a screen which can change in a range faster than 1-2 seconds not to miss the screen to check for.

Reply:
This still applies to the current version

Reply:
No it does not. The `check_screen` is executed *ONLY* when the previously existing `assert_screen` succeeds. And that has a 3600s timeout and is not supposed to return often or soon. If the `check_screen` succeeds, it means the installation is still ongoing so we go back to the assert_screen.

One can tell that the `check_screen` is not happening all the time because there is not an avalanche of screenshots, like my previous approach.

Reply:
Sorry but regardless I don't want to approve as long as there is a `do { … } while (check_screen);` without any additional abort condition. A for loop with a sane ending limit would be less worse

Reply:
Well, in general there's the  2h hard limit, I think that's good enough abort condition, isn't it? if it fails inside, then it's a problem yet unforseeable... at least from what I can see atm

Reply:
`check_screen` "Returns matched needle or undef if timeout is hit. Default timeout is 0s."

Why is it invalid as a terminating condition for a `do{...}while(cond)` loop?

Reply:
I reread the postfail hook comment and I understand now. `assert_screen` is still not a good choice because this is meant to fail without causing the test to fail. Would you accept a "manual" timeout?

Reply:
I don't know what you mean with "manual" timeout. I suggested a for loop, like in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/x11/oomath.pm#L30

Reply:
I committed the second you posted that xD